<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from robot.xacro                    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="UMARU" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="cyan">
    <color rgba="0 255 255 1"/>
  </material>
  <material name="yellow">
    <color rgba="255 255 0 1"/>
  </material>
  <link name="body">
    <visual>
      <geometry>
        <box size="0.117  0.147 0.176124"/>
      </geometry>
      <origin rpy="0 0 0"/>
      <material name="cyan"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.117  0.147 0.176124"/>
      </geometry>
      <origin rpy="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.8"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0078942995064" ixy="0" ixz="0" iyy="0.006706299506400001" iyz="0" izz="0.005294699999999999"/>
    </inertial>
  </link>
  <link name="right_shoulder_horizontal">
    <visual>
      <geometry>
        <box size="0.065 0.055 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 -0.0275 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.065 0.055 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 -0.0275 0"/>
    </collision>
    <inertial>
      <mass value="0.07"/>
      <origin xyz="0 -0.0275 0"/>
      <inertia ixx="2.2895833333333332e-05" ixy="0" ixz="0" iyy="2.989583333333334e-05" iyz="0" izz="4.229166666666667e-05"/>
    </inertial>
  </link>
  <link name="left_shoulder_horizontal">
    <visual>
      <geometry>
        <box size="0.065 0.055 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0.0275 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.065 0.055 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0.0275 0"/>
    </collision>
    <inertial>
      <mass value="0.07"/>
      <origin xyz="0 0.0275 0"/>
      <inertia ixx="2.2895833333333332e-05" ixy="0" ixz="0" iyy="2.989583333333334e-05" iyz="0" izz="4.229166666666667e-05"/>
    </inertial>
  </link>
  <link name="right_shoulder_vertical">
    <visual>
      <geometry>
        <box size="0.065 0.03 0.02"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.01"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.065 0.03 0.02"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.01"/>
    </collision>
    <inertial>
      <mass value="0.07"/>
      <origin xyz="0 0 -0.01"/>
      <inertia ixx="7.583333333333333e-06" ixy="0" ixz="0" iyy="2.6979166666666674e-05" iyz="0" izz="2.989583333333334e-05"/>
    </inertial>
  </link>
  <link name="left_shoulder_vertical">
    <visual>
      <geometry>
        <box size="0.065 0.03 0.02"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.01"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.065 0.03 0.02"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.01"/>
    </collision>
    <inertial>
      <mass value="0.07"/>
      <origin xyz="0 0 -0.01"/>
      <inertia ixx="7.583333333333333e-06" ixy="0" ixz="0" iyy="2.6979166666666674e-05" iyz="0" izz="2.989583333333334e-05"/>
    </inertial>
  </link>
  <link name="right_arm">
    <visual>
      <geometry>
        <box size="0.038 0.0402 0.156"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.078"/>
      <material name="cyan"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.038 0.0402 0.156"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.078"/>
    </collision>
    <inertial>
      <mass value="0.12"/>
      <origin xyz="0 0 -0.078"/>
      <inertia ixx="0.0002595204" ixy="0" ixz="0" iyy="0.00025780000000000003" iyz="0" izz="3.06004e-05"/>
    </inertial>
  </link>
  <link name="left_arm">
    <visual>
      <geometry>
        <box size="0.038 0.0402 0.156"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.078"/>
      <material name="cyan"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.038 0.0402 0.156"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.078"/>
    </collision>
    <inertial>
      <mass value="0.12"/>
      <origin xyz="0 0 -0.078"/>
      <inertia ixx="0.0002595204" ixy="0" ixz="0" iyy="0.00025780000000000003" iyz="0" izz="3.06004e-05"/>
    </inertial>
  </link>
  <link name="right_hand">
    <visual>
      <geometry>
        <box size="0.0276 0.0505 0.162"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.081"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.0276 0.0505 0.162"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.081"/>
    </collision>
    <inertial>
      <mass value="0.08"/>
      <origin xyz="0 0 -0.081"/>
      <inertia ixx="0.00019196166666666668" ixy="0" ixz="0" iyy="0.00018003840000000001" iyz="0" izz="2.2080066666666672e-05"/>
    </inertial>
  </link>
  <link name="left_hand">
    <visual>
      <geometry>
        <box size="0.0276 0.0505 0.162"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.081"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.0276 0.0505 0.162"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.081"/>
    </collision>
    <inertial>
      <mass value="0.08"/>
      <origin xyz="0 0 -0.081"/>
      <inertia ixx="0.00019196166666666668" ixy="0" ixz="0" iyy="0.00018003840000000001" iyz="0" izz="2.2080066666666672e-05"/>
    </inertial>
  </link>
  <link name="right_hip">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="left_hip">
    <inertial>
      <mass value="0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <!-- HIP TEST WITH EMPTY LINK -->
  <!-- <xacro:property name="hip_tall" value="0.162"/>
  <xacro:property name="hip_mass" value="0.120"/>
  <link name="right_hip">
    <visual>
      <geometry>
        <box size="${leg_width} ${leg_width} ${hip_tall}"/>
      </geometry>
      <origin xyz="0 0 ${- hip_tall/2}" rpy="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="${leg_width} ${leg_width} ${hip_tall}"/>
      </geometry>
      <origin xyz="0 0 ${- hip_tall/2}" rpy="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${hip_mass}"/>
      <origin xyz="0 0 ${- hip_tall/2}"/>
      <inertia ixx="${hip_mass * (leg_width**2 + hip_tall**2) / 12}" ixy="0" ixz="0" 
               iyy="${hip_mass * (leg_width**2 + hip_tall**2) / 12}" iyz="0" 
               izz="${hip_mass * (leg_width**2 + leg_width**2) / 12}" />
    </inertial>
  </link>
  <link name="left_hip">
    <visual>
      <geometry>
        <box size="${leg_width} ${leg_width} ${hip_tall}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 ${- hip_tall/2}"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="${leg_width} ${leg_width} ${hip_tall}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 ${- hip_tall/2}"/>
    </collision>
    <inertial>
      <mass value="${hip_mass}"/>
      <origin xyz="0 0 ${- hip_tall/2}"/>
      <inertia ixx="${hip_mass * (leg_width**2 + hip_tall**2) / 12}" ixy="0" ixz="0" 
               iyy="${hip_mass * (leg_width**2 + hip_tall**2) / 12}" iyz="0" 
               izz="${hip_mass * (leg_width**2 + leg_width**2) / 12}" />
    </inertial>
  </link>  -->
  <link name="left_upper_leg">
    <visual>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
    </collision>
    <inertial>
      <mass value="0.3"/>
      <origin xyz="0 0 -0.0715"/>
      <inertia ixx="0.0005712499999999998" ixy="0" ixz="0" iyy="0.0005712499999999998" iyz="0" izz="0.00012005000000000002"/>
    </inertial>
  </link>
  <link name="right_upper_leg">
    <visual>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
      <origin rpy="0 0  0" xyz="0 0 -0.0715"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
    </collision>
    <inertial>
      <mass value="0.3"/>
      <origin xyz="0 0 -0.0715"/>
      <inertia ixx="0.0005712499999999998" ixy="0" ixz="0" iyy="0.0005712499999999998" iyz="0" izz="0.00012005000000000002"/>
    </inertial>
  </link>
  <link name="right_hip_roll">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="stack_mass"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.00016333333333333334" ixy="0" ixz="0" iyy="0.00016333333333333334" iyz="0" izz="0.0002613333333333333"/>
    </inertial>
  </link>
  <link name="left_hip_roll">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="stack_mass"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.00016333333333333334" ixy="0" ixz="0" iyy="0.00016333333333333334" iyz="0" izz="0.0002613333333333333"/>
    </inertial>
  </link>
  <link name="left_lower_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
      <material name="cyan"/>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.3"/>
      <origin xyz="0 0 -0.0715"/>
      <inertia ixx="0.0005712499999999998" ixy="0" ixz="0" iyy="0.0005712499999999998" iyz="0" izz="0.00012005000000000002"/>
    </inertial>
  </link>
  <link name="right_lower_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
      <material name="cyan"/>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.0715"/>
      <geometry>
        <box size="0.049 0.049 0.143"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.3"/>
      <origin xyz="0 0 -0.0715"/>
      <inertia ixx="0.0005712499999999998" ixy="0" ixz="0" iyy="0.0005712499999999998" iyz="0" izz="0.00012005000000000002"/>
    </inertial>
  </link>
  <link name="left_foot">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.00155"/>
      <material name="cyan"/>
      <geometry>
        <box size="0.11 0.05 0.0031"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.00155"/>
      <geometry>
        <box size="0.11 0.05 0.0031"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.45"/>
      <origin xyz="0 0 -0.00155"/>
      <inertia ixx="9.411037500000002e-05" ixy="0" ixz="0" iyy="0.000454110375" iyz="0" izz="0.0005475"/>
    </inertial>
  </link>
  <link name="right_foot">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.00155"/>
      <material name="cyan"/>
      <geometry>
        <box size="0.11 0.05 0.0031"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.00155"/>
      <geometry>
        <box size="0.11 0.05 0.0031"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.45"/>
      <origin xyz="0 0 -0.00155"/>
      <inertia ixx="9.411037500000002e-05" ixy="0" ixz="0" iyy="0.000454110375" iyz="0" izz="0.0005475"/>
    </inertial>
  </link>
  <link name="right_ankle_roll">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="stack_mass"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.00016333333333333334" ixy="0" ixz="0" iyy="0.00016333333333333334" iyz="0" izz="0.0002613333333333333"/>
    </inertial>
  </link>
  <link name="left_ankle_roll">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="stack_mass"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.00016333333333333334" ixy="0" ixz="0" iyy="0.00016333333333333334" iyz="0" izz="0.0002613333333333333"/>
    </inertial>
  </link>
  <link name="neck">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
      <geometry>
        <box size="0.031 0.046 0.058"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.031 0.046 0.058"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.18"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="8.22e-05" ixy="0" ixz="0" iyy="6.487499999999999e-05" iyz="0" izz="4.6155e-05"/>
    </inertial>
  </link>
  <link name="head">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0275"/>
      <material name="cyan"/>
      <geometry>
        <box size="0.02 0.092 0.055"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0275"/>
      <geometry>
        <box size="0.02 0.092 0.055"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.05"/>
      <origin xyz="0 0 0.0275"/>
      <inertia ixx="4.7870833333333336e-05" ixy="0" ixz="0" iyy="1.4270833333333334e-05" iyz="0" izz="3.693333333333333e-05"/>
    </inertial>
  </link>
  <!-- Axis and limit might need changes -->
  <!-- JOINT -->
  <joint name="R_SHO_PITCH" type="revolute">
    <!-- JOINT_ID = 1, Pybullet = 0-->
    <axis xyz="0 -1 0"/>
    <parent link="body"/>
    <child link="right_shoulder_horizontal"/>
    <origin xyz="0 -0.0735 0.073062"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="L_SHO_PITCH" type="revolute">
    <!-- JOINT_ID = 2, Pybullet = 4 -->
    <axis xyz="0 -1 0"/>
    <parent link="body"/>
    <child link="left_shoulder_horizontal"/>
    <origin xyz="0 0.0735 0.073062"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="R_SHO_ROLL" type="revolute">
    <!-- JOINT_ID = 3, Pybullet = 1 -->
    <limit effort="1000.0" lower="0.0" upper="1.5707963267948966" velocity="0.5"/>
    <axis xyz="-1 0 0"/>
    <parent link="right_shoulder_vertical"/>
    <child link="right_arm"/>
    <origin xyz="0 0 -0.01"/>
  </joint>
  <joint name="L_SHO_ROLL" type="revolute">
    <!-- JOINT_ID = 4 -->
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0.0" upper="1.5707963267948966" velocity="0.5"/>
    <parent link="left_shoulder_vertical"/>
    <child link="left_arm"/>
    <origin xyz="0 0 -0.01"/>
  </joint>
  <joint name="R_ELBOW" type="revolute">
    <!-- JOINT_ID = 5 -->
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-4.71238898038469" upper="4.71238898038469" velocity="0.5"/>
    <parent link="right_arm"/>
    <child link="right_hand"/>
    <origin xyz="0 0 -0.156"/>
  </joint>
  <joint name="L_ELBOW" type="revolute">
    <!-- JOINT_ID = 6 -->
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-4.71238898038469" upper="4.71238898038469" velocity="0.5"/>
    <parent link="left_arm"/>
    <child link="left_hand"/>
    <origin xyz="0 0 -0.156"/>
  </joint>
  <joint name="R_HIP_ROLL" type="revolute">
    <!-- JOINT_ID = 7-->
    <parent link="right_hip"/>
    <child link="right_hip_roll"/>
    <axis xyz="-1 0 0"/>
    <origin xyz="0 0 0"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="L_HIP_ROLL" type="revolute">
    <!-- JOINT_ID = 8-->
    <parent link="left_hip"/>
    <child link="left_hip_roll"/>
    <axis xyz="1 0 0"/>
    <origin xyz="0 0 0"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="R_HIP_YAW" type="revolute">
    <!-- JOINT_ID = 9 -->
    <parent link="body"/>
    <child link="right_hip"/>
    <origin xyz="0 -0.040833333333333326 -0.088062"/>
    <axis xyz="0 0 -1"/>
    <limit effort="1000.0" lower="-1.5707963267948966" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="L_HIP_YAW" type="revolute">
    <!-- JOINT_ID = 10 -->
    <parent link="body"/>
    <child link="left_hip"/>
    <origin xyz="0 0.040833333333333326 -0.088062"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.5707963267948966" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="R_HIP_PITCH" type="revolute">
    <!-- JOINT_ID = 11-->
    <parent link="right_hip_roll"/>
    <child link="right_upper_leg"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0 0 0"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="L_HIP_PITCH" type="revolute">
    <!-- JOINT_ID = 12-->
    <parent link="left_hip_roll"/>
    <child link="left_upper_leg"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0 0 0"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="R_KNEE" type="revolute">
    <!-- JOINT_ID = 13 -->
    <parent link="right_upper_leg"/>
    <child link="right_lower_leg"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 0 -0.143"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="L_KNEE" type="revolute">
    <!-- JOINT_ID = 14 -->
    <parent link="left_upper_leg"/>
    <child link="left_lower_leg"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 0 -0.143"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="R_ANK_ROLL" type="revolute">
    <!-- JOINT_ID = 15 -->
    <parent link="right_lower_leg"/>
    <child link="right_ankle_roll"/>
    <axis xyz="-1 0 0"/>
    <origin xyz="0 0 -0.13299999999999998"/>
    <limit effort="1000.0" lower="0" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="L_ANK_ROLL" type="revolute">
    <!-- JOINT_ID = 16 -->
    <parent link="left_lower_leg"/>
    <child link="left_ankle_roll"/>
    <axis xyz="1 0 0"/>
    <origin xyz="0 0 -0.13299999999999998"/>
    <limit effort="1000.0" lower="0" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="R_ANK_PITCH" type="revolute">
    <!-- JOINT_ID = 17 -->
    <parent link="right_ankle_roll"/>
    <child link="right_foot"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0 0 -0.02"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
    <dynamics damping="1000" friction="1000"/>
  </joint>
  <joint name="L_ANK_PITCH" type="revolute">
    <!-- JOINT_ID = 18 -->
    <parent link="left_ankle_roll"/>
    <child link="left_foot"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0 0 -0.02"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="HEAD_PAN" type="revolute">
    <!-- JOINT_ID = 19 -->
    <parent link="body"/>
    <child link="neck"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0 0 0.117062"/>
    <limit effort="1000.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.5"/>
  </joint>
  <joint name="HEAD_TILT" type="revolute">
    <!-- JOINT_ID = 20 -->
    <parent link="neck"/>
    <child link="head"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0.0155 0 0"/>
    <limit effort="1000.0" lower="-1.2566370614359172" upper="1.2566370614359172" velocity="0.5"/>
  </joint>
  <!--Fixed Joint-->
  <joint name="RIGHT_SHO" type="fixed">
    <parent link="right_shoulder_horizontal"/>
    <child link="right_shoulder_vertical"/>
    <origin xyz="0 -0.0275 -0.015"/>
  </joint>
  <joint name="LEFT_SHO" type="fixed">
    <parent link="left_shoulder_horizontal"/>
    <child link="left_shoulder_vertical"/>
    <origin xyz="0 0.0275 -0.015"/>
  </joint>
</robot>
